hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/webapp/FederationInterceptorREST.java

```diff
@@ -761,7 +761,7 @@ public AppsInfo getApps(HttpServletRequest hsr, String stateQuery,
     });
 
     if (apps.getApps().isEmpty()) {
-      return null;
+      return apps;
```

Comment:
Is this what we want?

Reply:
Thank you very much for your help in reviewing the code! I was doing the integration test of YARN Federation recently and accidentally mixed some code into YARN-8862. I will fix this part of the code.